server/handlers.go

```diff
@@ -115,6 +115,36 @@ func (h *healthChecker) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "Health check passed in %s", t)
 }
 
+func (s *Server) tokenValidHandler(w http.ResponseWriter, r *http.Request) {
+	refreshCode := r.PostFormValue("refresh_token")
+	if refreshCode == "" {
+		s.tokenErrHelper(w, errInvalidRequest, "No refresh token in request.", http.StatusBadRequest)
+		return
+	}
+	token := new(internal.RefreshToken)
+	if err := internal.Unmarshal(refreshCode, token); err != nil {
+		// For backward compatibility, assume the refresh_token is a raw refresh token ID
+		// if it fails to decode.
+		//
+		// Because refresh_token values that aren't unmarshable were generated by servers
+		// that don't have a Token value, we'll still reject any attempts to claim a
+		// refresh_token twice.
+		token = &internal.RefreshToken{RefreshId: refreshCode, Token: ""}
+	}
+
+	refresh, err := s.storage.GetRefresh(token.RefreshId)
+	if err != nil {
+		s.tokenErrHelper(w, errServerError, "Refresh token not Found", http.StatusInternalServerError)
+		return
+	}
+	currTime := time.Now()
+	diff := currTime.Sub(refresh.LastUsed)
+	if diff.Hours() >= 1 {
```

Comment:
1 can be assigned to a global variable use variable here, later we could think of using it from config file too.